src/main/java/com/internet/shop/dao/impl/ShoppingCartJdbcDaoImpl.java

```diff
@@ -0,0 +1,153 @@
+package com.internet.shop.dao.impl;
+
+import com.internet.shop.dao.ShoppingCartDao;
+import com.internet.shop.dao.util.ConnectionUtil;
+import com.internet.shop.exceptions.DataProcessingException;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.models.Product;
+import com.internet.shop.models.ShoppingCart;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ShoppingCartJdbcDaoImpl implements ShoppingCartDao {
+    @Override
+    public Optional<ShoppingCart> getByUser(Long id) {
+        String query = "SELECT * FROM shopping_carts WHERE user_id = ? AND deleted = false";
+        ShoppingCart shoppingCart = new ShoppingCart();
+        try (PreparedStatement statement = ConnectionUtil.getConnection().prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                shoppingCart.setId(resultSet.getLong("cart_id"));
+                shoppingCart.setUserId(resultSet.getLong("user_id"));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find ShoppingCart by user ID: " + id, e);
+        }
+        getShoppingCartProducts(shoppingCart);
+        return Optional.of(shoppingCart);
```

Comment:
in this method you will never return optional empty, but we should do that if cart was not found